cclib/parser/turbomoleparser.py

```diff
@@ -869,18 +869,29 @@ def extract(self, inputfile, line):
         #              FOUND DFT-FLAG !
         #
         # For plain HF (not DFT) the "FOUND DFT-FLAG !" will be missing.
-        if line.strip() in ("CI SINGLES SINGLET-CALCULATION (TAMM-DANCOFF-APPROXIMATION)", "CI SINGLES TRIPLET-CALCULATION (TAMM-DANCOFF-APPROXIMATION)"):
-            self.metadata['excited_states_method'] = "CIS"
-        
-        elif line.strip() in ("RPA SINGLET-EXCITATION-CALCULATION", "RPA TRIPLET-EXCITATION-CALCULATION"):
-            self.metadata['excited_states_method'] = "RPA"
-            
-        elif line.strip() == "FOUND DFT-FLAG !":
-            if self.metadata['excited_states_method'] == "CIS":
-                self.metadata['excited_states_method'] = "TDA"
-            
+        if line.strip() in (
+            "CI SINGLES SINGLET-CALCULATION (TAMM-DANCOFF-APPROXIMATION)",
+            "CI SINGLES TRIPLET-CALCULATION (TAMM-DANCOFF-APPROXIMATION)",
+            "RPA SINGLET-EXCITATION-CALCULATION",
+             "RPA TRIPLET-EXCITATION-CALCULATION"
+        ):
+            line = next(inputfile)
+            line = next(inputfile)
+            if line.strip() == "FOUND DFT-FLAG !":
+                # TD-DFT and TDA
+                if "CI SINGLES" in line:
+                    self.metadata['excited_states_method'] = "TDA"

```

Comment:
Ok so something like:
```
tda_type = "TDA"
...
self.metadata['excited_states_method'] = tda_type
```

I tend not to go for this too much, I find it can result in a surprising amount of duplication. It also feels a bit too close to that bonkers old style of doing things like:
```
one = "one"
two = "two"
```
As we're implementing across a bunch of parsers though it could be useful perhaps? I'll make the change and see how you feel.

Reply:
Ah sorry I think I get what you mean know, like:
```
if line.strip() == "FOUND DFT-FLAG !":
    # TD-DFT and TDA
    if "CI SINGLES" in line:
        method = "TDA"
        
    else:
        method = "TD-DFT"
    
else:                
    # RPA and CIS
    if "CI SINGLES" in line:
        method = "CIS"
        
    else:
        method = "RPA"
        
self.metadata['excited_states_method'] = method
```

Reply:
Yes, like this.  Even doing it this way isn't always helpful or important.  It makes it easier to isolate where `excited_states_method` is set, and `method` is now a proper code object rather than just having a bunch of bare string constants hanging out.